Book Paging not working correctly

### Skript/Server Version

```markdown
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript] Server Version: 1.21.11-132-c5eb079 (MC: 1.21.11)
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript] Skript Version: 2.15.4 (skriptlang-github)
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript] Installed Skript Addons: 
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - skript-gui v1.3.2 (https://github.com/APickledWalrus/skript-gui)
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - skript-particle v1.4.1 (https://github.com/sovdeeth/skript-particle)
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - DiSky v4.27.0
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - skript-reflect v2.6.3 (https://github.com/SkriptLang/skript-reflect)
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - skript-worldedit v1.1.4
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - SkBee v3.25.1 (https://github.com/ShaneBeee/SkBee)
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - skript-citizens v1.1.0
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - skript-worldguard v1.0.1 (https://github.com/SkriptLang/skript-worldguard)
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript] Installed dependencies: 
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - Vault v1.7.3-b131
Jul  4 09:04:52:  [09:04:52 INFO]: [Skript]  - WorldGuard v7.0.15+2347-5fafea6
```

### Bug Description

When creating books, you would normally create new pages like this:
```
set {_book} to written book
set book page 1 of {_book} to "This is page 1"
set book page 2 of {_book} to "This is page 2"
open book {_book} to player
```
Somehow, this is no longer working. When executing a command that does this, Skript gives back an internal error, telling the player that the "page 1" is incorrect since the book has 0 pages.
I already tried starting with page 0 instead but the same error comes up. 
A solution I found is by creating all pages within a single line like this:
```
set {_book} to written book
set pages of {_book} to "This is page 1", "This is page 2"
open book {_book} to player
```


### Expected Behavior

Expected behaviour is to just open the book to the player when creating it like this.

### Steps to Reproduce

Copy the Skinfo and execute a command like this:
```
command /booktest:
    trigger:
        set {_book} to written book
        set book page 1 of {_book} to "This is page 1"
        set book page 2 of {_book} to "This is page 2"
        open book {_book} to player
```

### Errors or Screenshots

```
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! 
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! [Skript] Severe Error:
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! 
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! An unexpected error occurred with Skript. This issue is likely not your fault.
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! You are using some plugins that alter how Skript works (addons).
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Full list of addons:
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! skript-citizens v1.1.0
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! skript-gui v1.3.2 (https://github.com/APickledWalrus/skript-gui)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! SkBee v3.25.1 (https://github.com/ShaneBeee/SkBee)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! skript-particle v1.4.1 (https://github.com/sovdeeth/skript-particle)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! skript-reflect v2.6.3 (https://github.com/SkriptLang/skript-reflect)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! CraftEngine v0.0.67
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! DiSky v4.27.0
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! skript-worldedit v1.1.4
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! skript-worldguard v1.0.1 (https://github.com/SkriptLang/skript-worldguard)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Sktrace v0.1.1
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! We could not identify related addons, it might also be a Skript issue.
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Try temporarily removing the listed plugins one by one to identify the cause.
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! If removing a plugin resolves the issue, please report the problem to the plugin developer.
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! 
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Stack trace:
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Caused by: java.lang.IllegalArgumentException: Invalid page number (0/0)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at com.google.common.base.Preconditions.checkArgument(Preconditions.java:301)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at org.bukkit.craftbukkit.inventory.CraftMetaBookSigned.page(CraftMetaBookSigned.java:519)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//org.skriptlang.skript.bukkit.item.book.elements.expressions.ExprBookPages.change(ExprBookPages.java:132)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.effects.EffChange.execute(EffChange.java:327)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.lang.Effect.run(Effect.java:42)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:40)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:67)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:33)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.SkriptEventHandler.lambda$execute$0(SkriptEventHandler.java:169)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.SkriptEventHandler.lambda$execute$1(SkriptEventHandler.java:180)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.util.Task.callSync(Task.java:212)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.util.Task.callSync(Task.java:196)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.SkriptEventHandler.execute(SkriptEventHandler.java:178)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.SkriptEventHandler.check(SkriptEventHandler.java:111)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at Skript-2.15.4.jar//ch.njol.skript.SkriptEventHandler$PriorityListener.lambda$new$0(SkriptEventHandler.java:36)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:71)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:131)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl$1.performInteraction(ServerGamePacketListenerImpl.java:2834)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl$1.onInteraction(ServerGamePacketListenerImpl.java:2891)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.network.protocol.game.ServerboundInteractPacket$InteractionAction.dispatch(ServerboundInteractPacket.java:147)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.network.protocol.game.ServerboundInteractPacket.dispatch(ServerboundInteractPacket.java:98)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl.handleInteract(ServerGamePacketListenerImpl.java:2824)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.network.protocol.game.ServerboundInteractPacket.handle(ServerboundInteractPacket.java:80)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.network.protocol.game.ServerboundInteractPacket.handle(ServerboundInteractPacket.java:16)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.network.PacketProcessor$ListenerAndPacket.handle(PacketProcessor.java:99)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.network.PacketProcessor.executeSinglePacket(PacketProcessor.java:33)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.runAllTasksAtTickStart(MinecraftServer.java:1180)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.processPacketsAndTick(MinecraftServer.java:1666)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1335)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:388)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:1583)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! 
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Skript: 2.15.4 (latest)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     Flavor: skriptlang-github
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#!     Date: 12:17:17.572615
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Bukkit: 1.21.11-R0.1-SNAPSHOT
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Minecraft: 1.21.11
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Java: 21.0.9 (OpenJDK 64-Bit Server VM 21.0.9+10-Ubuntu-124.04)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! OS: Linux amd64 6.8.0-85-generic
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! 
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Server platform: Paper
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! 
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Current node: null
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Current item: set page [[integer:0]] of {_book} (as ch.njol.skript.aliases.ItemType) to ("&oOctober 3rd&0%new line%I don't know why I'm writing this. Maybe because it feels like the only thing left I can control.%new line%He's been following me for three days now. I first noticed him near the mill, just standing there, watching. When I turned around, he was gone." >> ch.njol.skript.lang.util.ConvertedExpression$$Lambda/0x00007ac59bc43ef0@448f2dc8: ConverterInfo{from=class java.lang.String,to=interface net.kyori.adventure.text.Component,converter=org.skriptlang.skript.bukkit.text.TextModule$$Lambda/0x00007ac59aa6b300@52a78700,flags=0})
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Current trigger: rightclick on item frame or glow item frame (rightclick on ([[entitydata:item frame]] or [[entitydata:glow item frame]])[class ch.njol.skript.entity.EntityData]) (missions/thecipher.sk, line 16)
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Thread: Server thread
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Language: english
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! Link parse mode: DISABLED
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! End of Error.
Jul  4 09:09:27:  [09:09:27 ERROR]: #!#! 
```

### Other

_No response_

### Agreement

- [x] I have read the guidelines above and affirm I am following them with this report.

In case of the error shown right here, I tried starting with book page 0. When starting with book page 1, the error should look something like this:
```
Caused by: java.lang.IllegalArgumentException: Invalid page number (1/0)
```

--- a/src/main/java/org/skriptlang/skript/bukkit/item/book/elements/expressions/ExprBookPages.java
+++ b/src/main/java/org/skriptlang/skript/bukkit/item/book/elements/expressions/ExprBookPages.java
@@ -22,7 +22,7 @@ import org.skriptlang.skript.registration.SyntaxInfo;
 import org.skriptlang.skript.registration.SyntaxRegistry;
 
 import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.List;
 
 @Name("Book Pages")
@@ -104,14 +104,17 @@ public class ExprBookPages extends SimpleExpression<Component> {
 
 	@Override
 	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
-		int pageNumber = isAllPages() ? -1 : this.pageNumber.getOptionalSingle(event).orElse(-1);
-		List<Component> newPages = delta == null ? Collections.emptyList() : new ArrayList<>(delta.length);
+		int pageNumber = -1;
+		if (!isAllPages()) {
+			pageNumber = this.pageNumber.getOptionalSingle(event).orElse(-1);
+			if (pageNumber <= 0) {
+				return;
+			}
+		}
+		List<Component> newPages = delta == null ? List.of() : new ArrayList<>(delta.length);
 		if (delta != null) {
 			for (Object page : delta) {
 				newPages.add((Component) page);
-			}
-			if (pageNumber != -1 && newPages.isEmpty()) { // not setting this page to anything, exit early
-				return;
 			}
 		}
 
@@ -129,14 +132,29 @@ public class ExprBookPages extends SimpleExpression<Component> {
 				}
 			} else {
 				switch (mode) {
-					case SET -> bookMeta.page(pageNumber, newPages.getFirst());
+					case SET -> {
+						if (pageNumber > bookMeta.getPageCount()) {
+							Component[] pages = new Component[pageNumber - bookMeta.getPageCount()];
+							Arrays.fill(pages, Component.empty());
+							bookMeta.addPages(pages);
+						}
+						bookMeta.page(pageNumber, newPages.getFirst());
+					}
 					case DELETE -> {
+						if (pageNumber > bookMeta.getPageCount()) {
+							break;
+						}
 						List<Component> pages = new ArrayList<>(bookMeta.pages());
-						pages.remove(pageNumber);
+						pages.remove(pageNumber - 1);
 						//noinspection ResultOfMethodCallIgnored - modifies in place despite contract
 						bookMeta.pages(pages);
 					}
-					case RESET -> bookMeta.page(pageNumber, Component.empty());
+					case RESET -> {
+						if (pageNumber > bookMeta.getPageCount()) {
+							continue;
+						}
+						bookMeta.page(pageNumber, Component.empty());
+					}
 					default -> throw new IllegalStateException();
 				}
 			}

--- a/src/test/skript/tests/bukkit/item/book/ExprBookPages.sk
+++ b/src/test/skript/tests/bukkit/item/book/ExprBookPages.sk
@@ -1,0 +1,22 @@
+test "ExprBookPages":
+	set {_i} to a written book
+
+	# all pages
+	assert {_i}'s book pages are not set with "non-existent book pages failed"
+	set pages of {_i} to "&aPages testing" and "test 2"
+	assert size of {_i}'s book pages is 2 with "failed to match exact pages size"
+	assert page 2 of {_i} is "test 2" with "failed to compare page 2"
+	add "test 3" to the pages of {_i}
+	assert page 3 of {_i} is "test 3" with "failed to compare page 3"
+	clear pages of {_i}
+	assert {_i}'s book pages are not set with "non-existent book pages failed 2"
+
+	# specific pages
+	set page 3 of {_i} to "test 3"
+	assert page 3 of {_i} is "test 3" with "failed to set page 3"
+	assert size of {_i}'s book pages is 3 with "failed to add correct number of pages"
+	delete page 2 of {_i}
+	# page 3 becomes page 2
+	assert page 2 of {_i} is "test 3" with "failed to remove page 2"
+	reset page 2 of {_i}
+	assert page 2 of {_i} is "" with "failed to reset page 2"

--- a/src/test/skript/tests/syntaxes/expressions/ExprBookPages.sk
+++ b/src/test/skript/tests/syntaxes/expressions/ExprBookPages.sk
@@ -1,6 +1,0 @@
-test "book pages":
-	set {_i} to a written book
-	assert {_i}'s book pages is not set with "non-existent book pages failed 2"
-	set pages of {_i} to "&aPages testing" and "test 2"
-	assert size of {_i}'s book pages is 2 with "failed to match exact pages size"
-	assert page 2 of {_i} is "test 2" with "failed to compare page 2"

